store/sqlstore/emoji_store.go

```diff
@@ -63,6 +63,31 @@ func (es SqlEmojiStore) Save(emoji *model.Emoji) (*model.Emoji, *model.AppError)
 	return emoji, nil
 }
 
+// GetBy returns one active (not deleted) emoji, found by any one column (what/key).
+func (es SqlEmojiStore) GetBy(what string, key interface{}) (*model.Emoji, *model.AppError) {
+	var emoji *model.Emoji
+
+	err := es.GetReplica().SelectOne(&emoji,
+		`SELECT
```

Comment:
Worth leveraging squirrel to build the query instead? (See some of our other sql stores for examples.)

Reply:
I don't mind changing that but I think it's not worth it, with the store being small and queries being simple.